### ISSUE TITLE
Integrate Nexus into the mvp demo

### ISSUE THREAD

**Issue by Keith-CY** [ckb-js/kuai#101]
The MVP DApp is expected to be [work with MetaMask](https://github.com/ckb-js/kuai/issues/54#issuecomment-1365630586) at this early stage.

Meanwhile, a browser extension wallet of Nervos CKB, named Nexus(https://github.com/ckb-js/nexus) is under development.

Once Nexus is ready for tests, we will try to support it in the MVP DApp.

**Comment by Keith-CY** [comment]
We're considering integrating Nexus into the Kuai MVP DApp(https://github.com/ckb-js/kuai/issues/54), any documents would you recommend for us to start? @homura 

Online demo: https://kuai-mvp-dapp-ui.vercel.app/

**Comment by homura** [comment]
Nexus don't currently have a friendly document, only a [pre-design](https://github.com/ckb-js/nexus/blob/pre-design/docs/design.md#javascript-provider) in the PR, we plan to provide a [demo dApp](https://github.com/zhb666/Nervos-DAO) of Nervos DAO to show how to use Nexus. Nexus MVP will probably be released around mid-March

**Comment by Keith-CY** [comment]
> Nexus don't currently have a friendly document, only a [pre-design](https://github.com/ckb-js/nexus/blob/pre-design/docs/design.md?rgh-link-date=2023-02-01T06%3A56%3A36Z#javascript-provider) in the PR, we plan to provide a [demo dApp](https://github.com/zhb666/Nervos-DAO) of Nervos DAO to show how to use Nexus. Nexus MVP will probably be released around mid-March

Cool, looking forward to it

**Comment by Keith-CY** [comment]
Tmp ddl: 03/08

**Comment by Keith-CY** [comment]
Demo repo: https://github.com/zhangyouxin/demo-nexus/

**Comment by Keith-CY** [comment]
Connect to nexus: https://github.com/zhangyouxin/demo-nexus/blob/5b25da92df19487511b0e87a293ec660579d169f/pages/index.tsx#L231-L236
Create next receiving address: https://github.com/zhangyouxin/demo-nexus/blob/5b25da92df19487511b0e87a293ec660579d169f/pages/index.tsx#L243-L250
Sign transaction of full ownership lock: https://github.com/zhangyouxin/demo-nexus/blob/5b25da92df19487511b0e87a293ec660579d169f/pages/index.tsx#L180-L183

**Comment by Keith-CY** [comment]
There're 3 obstacles on integrating nexus

1. failed to send a transaction within the demo
 

https://user-images.githubusercontent.com/7271329/225075900-7dd133fc-d69e-49dc-b383-5d47a6cf9892.mov

2. APIs used in the demo are out of date, but new APIs are documented(https://github.com/ckb-js/nexus/blob/main/docs/rpc.md), so it won't block the development, just not worth doing it now.

3. The extension doesn't work for unknown reasons


https://user-images.githubusercontent.com/7271329/225077456-03288d4e-5029-4dd0-97dc-65d00a1519f7.mov

    1. The popup doesn't show on click
    2. `nexus wallet not found` is prompted in console
    3. During integration with kuai MVP, the popup is blank

https://user-images.githubusercontent.com/7271329/225079179-2761bddc-9b8b-47cf-bdaf-eb1809186e34.mov

I will send this feedback to the nexus team tomorrow

**Comment by Keith-CY** [comment]
After some tries, I found the `nexus not found` happens after `refresh`, but will be fixed by `hard refresh`


https://user-images.githubusercontent.com/7271329/225084848-6153532d-c413-41fe-8c1d-e0c18c9e2461.mov

So the timing of detecting injection is improper

**Comment by homura** [comment]
hi, @Keith-CY could you please provide the following information to help debug

1. where did you download the Nexus?
2. if possible, can you provide the Nexus console output

<img width="357" alt="image" src="https://user-images.githubusercontent.com/7511174/225188392-744c3e3f-6127-4bc0-a167-c0c43556727c.png">

**Comment by Keith-CY** [comment]
> hi, @Keith-CY could you please provide the following information to help debug
> 
> 1. where did you download the Nexus?
> 2. if possible, can you provide the Nexus console output
> 
> <img alt="image" width="357" src="https://user-images.githubusercontent.com/7511174/225188392-744c3e3f-6127-4bc0-a167-c0c43556727c.png">

The extension was downloaded from https://github.com/zhangyouxin/demo-nexus/releases/tag/0.1.0, which is prompted in the demo
<img width="595" alt="image" src="https://user-images.githubusercontent.com/7271329/225202104-04115f38-dd4c-4286-a097-cc756c01c69a.png">

-----

The `inspect popup` makes the popup always on

https://user-images.githubusercontent.com/7271329/225203194-139086aa-2568-4433-8c91-324f5172bb06.mov

**Comment by zhangyouxin** [comment]
According to the error message prompted when firing a transaction: `Incorrect password`, I doubt the nexus-wallet-chrome-extension is a deprecated version, so I built another package using the latest code and uploaded it here: https://github.com/zhangyouxin/demo-nexus/releases/tag/0.1.1, could you try this one pls?

**Comment by Keith-CY** [comment]
Found another issue blocking the interaction
https://github.com/ckb-js/nexus/issues/188

**Comment by Keith-CY** [comment]
Front-end: https://github.com/Magickbase/kuai-mvp-dapp-ui/pull/13
Backend: https://github.com/ckb-js/kuai/pull/220
